roles/common/templates/pulp-operator-sa.yaml.j2

```diff
@@ -0,0 +1,21 @@
+apiVersion: v1
+{% if image_pull_secret|length > 0 and image_pull_secrets|length == 0 %}
+# image_pull_secret is deprecated in favor of image_pull_secrets
+imagePullSecrets:
+- name: {{ image_pull_secret }}
+{% if __dockercfg_secret is defined and __dockercfg_secret | length > 0%}
+- name: {{ __dockercfg_secret }}
+{% endif %}
+{% else %}
+imagePullSecrets:
+{% for secret in image_pull_secrets %}
+- name: {{ secret }}
+{% endfor %}
+{% if __dockercfg_secret is defined and __dockercfg_secret | length > 0 %}
+- name: {{ __dockercfg_secret }}
+{% endif %}
```

Comment:
this can be outside the if/else to avoid repetition

Reply:
Hi @fao89,
Thank you for reviewing it!
This is something that also caught my attention. My concern on putting it outside of the if/else is that I would also need to bring the `imagePullSecrets` field outside of the condition block (to avoid declaring it multiple times):
```
apiVersion: v1
imagePullSecrets:
{% if image_pull_secret|length > 0 and image_pull_secrets|length == 0 %}
# image_pull_secret is deprecated in favor of image_pull_secrets
- name: {{ image_pull_secret }}
{% else %}
{% for secret in image_pull_secrets %}
- name: {{ secret }}
{% endfor %}
{% if __dockercfg_secret is defined and __dockercfg_secret | length > 0%}
- name: {{ __dockercfg_secret }}
{% endif %}
```
The problem with this approach is when none of `image_pull_secret[s]` nor `__dockercfg_secret` is defined the `kubectl apply` operation will try something like:
```
$ kubectl get sa test -oyaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: test
  namespace: test
secrets:
- name: test-....

$ kubectl apply -f- <<EOF
apiVersion: v1
imagePullSecrets:
kind: ServiceAccount
metadata:
  name: test
  namespace: test
EOF
serviceaccount/test configured  <---------------

$ kubectl apply -f- <<EOF
apiVersion: v1
imagePullSecrets:
kind: ServiceAccount
metadata:
  name: test
  namespace: test
EOF
serviceaccount/test configured  <---------------
```
which seems to be modifying something in every execution (even though we are not trying to update anything).
I'll do some tests with the playbook to check if the k8s ansible module behaves in the same way as the `kubectl apply` command. 
Let me know if you agree with the above and have any suggestions.

Reply:
gotcha!

Reply:
I did some tests on minikube environment and it seems like ansible k8s module does not act like `kubectl apply`.
I'll do some tests on OCP env now just to make sure that everything works as expected.